doc/savefileformat.md

```diff
@@ -0,0 +1,419 @@
+# openage save format
+
+1.  [Questions](#1-questions)
+2.  [Overview](#2-overview)
+3.  [Header](#3-header)
+4.  [String Table](#4-string-table)
+5.  [Atrribute Table](#5-attribute-table)
+6.  [Game Information](#6-game-information)
+7.  [User Information](#7-user-information)
+8.  [Chunk Table](#8-chunk-table)
+9.  [Single Chunks](#9-single-chunks)
+10. [Triggers](#10-triggers)
+
+### 1 Questions
+
+Do we something like ingame time vs realtime if the game is played at a different speed?
+
+What do we want to track for stats?
+
+| category   | stats  |
+| ---------- | ------ |
+| ressources | collected food/wood/gold/stone, maximum res in the bank? |
+| units      | units/buildings killed/lost, units converted/lost to convertions, max units of each type |
+| map        | discovered, maximum simultanously viewable (discovered - fog of war) |
+| tech       | when was each tech/age researched? |
+| battle     | what is a battle? units killed per timeslot per close area 30 units died in 15 seconds in radius of 20 Tiles TODO: definitions, maybe add cost for unit for calculation of the size of battle |
+
+do we record timeseries? if yes fixed sizes(round robin database) or simple array snapshot per 10s/30s/60s/flexible?
+
+every minute we save unit count per player?
+
+crypto? multiplayer safe => server creates savefile, encrypts it, but one member of every team is needed to decrypt it and continue the game https://en.wikipedia.org/wiki/Secret_sharing
```

Comment:
does every player get the encrypted savefile or is it only stored on the server? the former is better for the (public) server (no harddrive space for stroing games needed), but leads to (1) file-integrity and (2) user authentity questions. (1) the players could together decrypt the file and change the gamestate. if everyone [or one of every team] is involved, no one will call the other ones cheater. but they could make a weird gamestate that breaks the engine by creating a weird logic (leading to remote code execution[admittedly, in the worst case]) or results in a DoS (creating 10^9 units or something).